sled-agent/src/rack_setup/service.rs

```diff
@@ -574,6 +583,15 @@ impl ServiceInner {
         .into_iter()
         .collect::<Result<_, SetupServiceError>>()?;
 
+        let dns_servers = DnsUpdater::new(
+            &config.az_subnet(),
+            self.log.new(o!("client" => "DNS")),
+        );
+        self.dns_servers
+            .set(dns_servers)
+            .map_err(|_| ())
+            .expect("DNS servers should only be set once");
```

Comment:
So as a heads up, I'm storing `dns_servers` in the struct because this is used in a follow-up PR: 
https://github.com/oxidecomputer/omicron/pull/1216/ 

In this patch, there's really no reason to store `dns_servers` in `self` ; I did this back-porting so subsequent patches could rely on these DNS records before the full RSS handoff.

If you feel strongly, I can refactor this PR, but I think I would need to end up changing it back in https://github.com/oxidecomputer/omicron/pull/1216/

Reply:
I do not feel strongly! Don't do extra work for this.